massmailing/src/main/resources/public/ts/model/Massmailing.ts

```diff
@@ -119,6 +122,28 @@ export class Massmailing {
             }
             student.opened = false;
         });
+
+        this.updateMassmailingCount();
+    }
+
+    public updateMassmailingCount(): void {
+        let count: number = 0;
+        this.students.forEach((student: MassmailingStudent) => {
+            let countStudent: number = 0;
+            student.relative.forEach((relative: IRelative) => {
+                if (relative.contact !== null) {
+                    countStudent += relative.selected ? 1 : 0;
+                }
+            });
+            if (this.isMultiple) {
+                let nbEvents: number = 0;
+                Object.keys(student.events).forEach((eventType: string) => nbEvents += student.events[eventType]);
+                countStudent *= nbEvents;
+            }
+            count += countStudent;
+        });
```

Comment:
this.students.map((student: MassmailingStudent) => student.relative.filter((relative: IRelative) => relative.contact !== null && relative.selected).length * this.isMultiple ?  Object.keys(student.events).map((eventType: string) => student.events[eventType]).reduce((a: number, b; number) => a + b, 0) : 1).reduce((a: number, b; number) => a + b, 0) 

Je ne sais pas si c'est plus court, test, sinon, laisse tomber !

Reply:
C'est pas énormément plus court et peut-être un peu moins lisible je trouve ^^